powerhome/playbook
# [CHMN-60] HaveIBeenPwned passphrase

#### Screens
No new visuals.
#### Breaking Changes
No, additive changes to the passphrase kit, with some refactoring. Other kits not affected. Passphrase kit not used yet (but will be soon)

Passphrase kit
  - add a new prop to enable use of https://haveibeenpwned.com/Passwords . This is a free api that you pass the first 5 characters of a potential passphrase sha-1 hash to, and it responds with ~500 sha-1 hashes that have the same first 5. If your hash is in the response, the passphrase you want to use is breached and shouldn't be used.

 - Add strength to the data attribute of the kit so a rails page can access that information and act on it. Currently, its just a callback in react so its hard to work with in rails.
 
- refactor use of "zxcvbn" into a hook

#### Runway Ticket URL

[CHMN-60](https://nitro.powerhrg.com/runway/backlog_items/CHMN-60)
#### How to test this

/kits/passphrase/react
/kits/passphrase/

See that the data attribute changes as you type
See that the API is hit when typing in the last example on each page.
#### Checklist:


*The normal release cut off deadline is 3p EDT each week. Please reach out to the release team if you have an urgent request or need a release off cycle.*
